src/main/java/com/gmail/yuramitryahin/dao/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,57 @@
+package com.gmail.yuramitryahin.dao;
+
+import com.gmail.yuramitryahin.exception.DataProcessingException;
+import com.gmail.yuramitryahin.lib.Dao;
+import com.gmail.yuramitryahin.model.MovieSession;
+import com.gmail.yuramitryahin.util.HibernateUtil;
+import java.time.LocalDate;
+import java.util.List;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            CriteriaBuilder builder = session.getCriteriaBuilder();
+            CriteriaQuery<MovieSession> query = builder.createQuery(MovieSession.class);
+            Root<MovieSession> root = query.from(MovieSession.class);
+            Predicate predicateId = builder.equal(root.get("id"), movieId);
```

Comment:
should movieSessionId in the DB be equal to `movieId` param?

Reply:
+